frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -301,10 +304,7 @@ pub mod pallet {
                 !RegisteredDapps::<T>::contains_key(&contract_id),
                 Error::<T>::AlreadyRegisteredContract
             );
-            ensure!(
-                Self::is_contract_valid(&contract_id),
-                Error::<T>::ContractIsNotValid
-            );
+            ensure!(contract_id.is_contract(), Error::<T>::ContractIsNotValid);
```

Comment:
You probably get these errors because you haven't imported the trait.

I might be wrong though.